src/commands/tasks/sync.ts

```diff
@@ -110,11 +83,11 @@ async function syncSingleTask(sdk:any, config: TaskConfig) {
   const file = await fs.readFile(uploadFilePath);
 
   /* Check if the function already exists */
-  const allFunctions = (await sdk.raw.get('/tasks/v1/functions')).data.data;
-  const myFunction = allFunctions.find((f:any) => f.name === config.name);
+  const rql = rqlBuilder().eq('name', config.name).build();
+  const myFunction = await taskRepository.functions.findFirst(sdk, { rql });
```

Comment:
After sleeping on it, this should be changed to `findByName` I wasn't eager to change any logic, but this is worse in both ways.

Reply:
This has been reverted to the following snipppet 

```JS
  /* Check if the function already exists */
  const allFunctions = await taskRepository.functions.find(sdk);
  const myFunction = allFunctions.find((f:any) => f.name === config.name);
  ```
  
  After some debugging, I realised the tasks service does not consider RQL for the request, the functions are resolved straight from AWS and spat back at us.